src/DynamoCore/Configuration/PreferenceSettings.cs

```diff
@@ -52,6 +52,8 @@ public class PreferenceSettings : NotificationObject, IPreferences, IRenderPreci
         private bool isBackgroundGridVisible;
         private bool disableTrustWarnings = false;
         private bool isNotificationCenterEnabled;
+        private bool isValid = true;
```

Comment:
why not just create an auto property, then you don't need to create the extra method below that is basically just a setter.

Reply:
> why not just create an auto property, then you don't need to create the extra method below that is basically just a getter.
> 
> You can make the setter internal and the getter public.

Yh I was thinking about thatbut kindly: per my understanding, the reason of this class is serves as a mapping from each property to a setting xml file, that's the reason Why I didn't add and extra property just for validation aproach, please your thoughts.